Falanx.Ast/Utils.fs

```diff
@@ -250,13 +252,23 @@ namespace Falanx.Ast
                 elif isListType instance.Type then None
                 // a property is a union case field \iff its declaring type is a proper subtype of the union type
                 elif instance.Type = propertyInfo.DeclaringType then None
-                else
-                    // create a dummy instance for declaring type to recover union case info
-                    let dummy = FormatterServices.GetUninitializedObject propertyInfo.DeclaringType
-                    let uci,_ = FSharpValue.GetUnionFields(dummy, propertyInfo.DeclaringType, true)
-                    match uci.GetFields() |> Array.tryFindIndex (fun p -> p = propertyInfo) with
-                    | Some i -> Some(instance, uci, i)
-                    | None -> None
+                else 
+                    match propertyInfo.DeclaringType with 
+                    | :? ProvidedTypeDefinition as puc -> 
+                        match puc.GetProperties() |> Seq.tryFind (fun p -> p.Name = propertyInfo.Name) with 
+                        | Some (pinfo) ->
+                            match pinfo.CustomAttributes |> Seq.tryFind (fun x -> x.AttributeType = typeof<CompilationMappingAttribute>) with 

```

Comment:
Some of the above hackery was because CustomAttributes were broken in providedTypes.fs upstream, which I since fixed.